CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Joel Winarske
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+cmake_minimum_required(VERSION 3.10.2)
+
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build, options are: Debug, Release, or MinSizeRel." FORCE)
+    message(STATUS "CMAKE_BUILD_TYPE not set, defaulting to Release.")
+endif()
+
+project(flutter-pi LANGUAGES C)
+
+message(STATUS "Generator .............. ${CMAKE_GENERATOR}")
+message(STATUS "Build Type ............. ${CMAKE_BUILD_TYPE}")
+
+if(NOT FLUTTER_ENGINE_LIBRARY)
```

Comment:
I think it'd be better to first search for the flutter engine library / header in the default library path / default include path. It's not necessary to download the engine artifacts when compiling on-target

Reply:
I'll look into creating a package for the engine build, such that the pattern matches the other libraries.  The expectation is you have to set FLUTTER_EGINE_LIBRARY if you're not building a desktop image.  It also provides a way to override when building as an external project, such as [I do](https://github.com/jwinarske/flutter_embedded/blob/master/cmake/rpi.cmake#L97) for this project.

So if you have the engine and header in your build folder:
`mkdir build && cd build && cmake .. -DFLUTTER_ENGINE_LIBRARY=. && make -j`